core/templates/dev/head/pages/exploration-player-page/services/audio-translation-language.service.ts

```diff
@@ -17,147 +17,172 @@
  * used for audio translations.
  */
 
-require('domain/utilities/browser-checker.service.ts');
-require('domain/utilities/language-util.service.ts');
-
-angular.module('oppia').factory('AudioTranslationLanguageService', [
-  'BrowserCheckerService', 'LanguageUtilService',
-  function(
-      BrowserCheckerService, LanguageUtilService) {
-    var _currentAudioLanguageCode = null;
-    var _allAudioLanguageCodesInExploration = null;
-    var _explorationLanguageCode = null;
-    var _automaticTextToSpeechEnabled = null;
-    var _languagesInExploration = [];
-
-    var attemptToSetAudioLanguageToExplorationLanguage = function() {
-      // We minimize the number of related languages, because we want to
-      // pick the language that is the most directly related to the exploration
-      // language. For example, this would prioritize Hindi over Hinglish
-      // if both were available as audio languages.
-      var numRelatedLanguages = Number.MAX_VALUE;
-      _allAudioLanguageCodesInExploration.forEach(function(audioLanguageCode) {
-        var relatedLanguageCodes =
-          LanguageUtilService.getLanguageCodesRelatedToAudioLanguageCode(
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { BrowserCheckerService } from
+  'domain/utilities/browser-checker.service';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AudioTranslationLanguageService {
+  constructor(private browserCheckerService: BrowserCheckerService,
+              private languageUtilService: LanguageUtilService) {}
+
+  _currentAudioLanguageCode: string = null;
+  _allAudioLanguageCodesInExploration: Array<string> = null;
+  _explorationLanguageCode: string = null;
+  _automaticTextToSpeechEnabled: boolean = null;
+  // TODO(#7165): Replace any with correct type.
+  _languagesInExploration: Array<any> = [];
+
+  attemptToSetAudioLanguageToExplorationLanguage(): void {
+    // We minimize the number of related languages, because we want to
+    // pick the language that is the most directly related to the exploration
+    // language. For example, this would prioritize Hindi over Hinglish
+    // if both were available as audio languages.
+    let numRelatedLanguages = Number.MAX_VALUE;
+    this._allAudioLanguageCodesInExploration.forEach((audioLanguageCode) => {
```

Comment:
Type for `audioLanguageCode`?

Reply:
Sorry for missing those...Fixed now.
Thanks!